client/src/pages/chart.jsx

```diff
@@ -91,7 +101,33 @@ class ChartComponent extends Component {
     this.updatePortfolioData();
   }
 
-  
+  runGetData(ticker) {
+    getData(this.state.end - 5097600, this.state.end, ticker, "30").then(({ data }) => {
+      this.setState({ data })
+      this.setState({ hasQueryData : data.hasData})
+      if (data.hasData)
+      {
+        this.setState({ mostRecentPrice : data.closes[data.closes.length - 1] });
```

Comment:
Why not put all this in a single `setState` call?

```
this.setState(prevState=> ({
  data,
  hasQueryData: data.hasData,
  mostRecentPrice: data.hasData? data.closes[data.closes.length - 1] : prevState.mostRecentPrice
}));

Reply:
Done